docs/tokens.md

```diff
@@ -3,137 +3,239 @@ id: tokens
+```javascript
+> GameItem.awardItem(playerAddress, "https://game.com/item-id-8u5h2m.json")
+7
+```
 
-You probably want to use a decimals of `18`, just like Ether, unless you have a special reason not to, so when you're minting tokens to people or transferring them around, you're actually sending the number `numTKN * 10^(decimals)`. So if I'm sending you `5` tokens using a token contract with 18 decimals, the method I'm executing actually looks like `transfer(yourAddress, 5 * 10^18)`.
+And the owner and metadata of each item queried:
+```javascript
+> GameItem.ownerOf(7)
+playerAddress
+> GameItem.tokenURI(7)
+"https://game.com/item-id-8u5h2m.json"
+```
 
-## ERC721
+This `tokenURI` should resolve to a JSON document that might look something like:
+
+```json
+{
+    "name": "Thor's hammer",
+    "description": "Mjölnir, the legendary hammer of the Norse god of thunder.",
+    "image": "https://game.com/item-id-8u5h2m.png",
+    "strength": 20
+}
+```
+
+For more information about the `tokenURI` metadata JSON Schema, check out the [ERC721 specification](https://eips.ethereum.org/EIPS/eip-721).
+
+_Note: you'll notice that the item's information is included in the metadata, but that information isn't on-chain! So a game developer could change the underlying metadata, changing the rules of the game! If you'd like to put all item information on-chain, you can extend ERC721 to do so (though it will be rather costly). You could also leverage IPFS to pin the tokenURI information, but these techniques are out of the scope of this overview guide._
+
+---
+
+# Advanced standards
+
+[ERC20](#erc20) and [ERC721](#erc721) (fungible and non-fungible assets, respectively) are the first two token contract standards to enjoy widespread use and adoption, but over time, multiple weak points of these standards were identified, as more advanced use cases came up.
 
-We've discussed how you can make a _fungible_ token using ERC20, but what if not all tokens are alike? This comes up in situations like company stock; some stock is common stock and some stock is investor shares, etc. It also comes up in a bunch of other places like in-game items, time, property, and so on.
+As a result, a multitude of new token standards were and are still being developed, with different tradeoffs between complexity, compatibility and ease of use. We'll explore some of those here.
 
-[ERC721](https://eips.ethereum.org/EIPS/eip-721) is a standard for representing ownership that is **non-fungible** aka, each token has unique properties.
+## ERC777
 
-Let's see what contracts OpenZeppelin provides for helping us work with ERC721:
+Like ERC20, ERC777 is a standard for [_fungible_ tokens](#different-kinds-of-tokens), and is focused around allowing more complex interactions when trading tokens, such as payment.
```

Comment:
What do you mean by "such as payment"?

Reply:
I meant payment via tokens (i.e. the receipt can execute code), similar to regular fallback functions.

Reply:
Ok, that's way too implicit. :stuck_out_tongue: However, I understand that it's hard to summarize the main goal(s) of ERC777, so I'm not sure what other thing we could say here.

Reply:
Two paragraphs later we say:

>and more generally brings tokens and Ether closer together by providing the equivalent of a `msg.value` field, but for tokens.

Do you think we should move that one here, and then go more in-depth there?

Reply:
Ah, I do like that explanation! I think it makes sense to use that one here.